lineapy/data/graph.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, nodes: List[Node], session_context: SessionContext):
                 (parent_id, node.id)
                 for node in nodes
                 for parent_id in node.parents()
+                if parent_id in set(self.ids.keys())
```

Comment:
In what scenarios will the parent_id be missing from self.ids.keys(). Should we throw an error or let it pass? What are the downstream implications?

Reply:
The `Graph.get_subgraph(nodes)` method needs this constraint to guarantee the return graph does not have extra nodes.